markup_tag_evaluation/tag_evaluation.py

```diff
@@ -2,68 +2,137 @@
+    num_inconsistent_tag_count: int
+    num_inconsistent_text: int
 
-    def average_character_difference(self) -> float:
-        return self.character_difference / max(1, self.number_of_tags)
+    num_tags_inconsistent_sentences: int
+    num_sentences: int
+    tgt_language: str
 
-    def inconsistent_sentences_percentage(self) -> float:
-        return self.number_of_inconsistent_sentences / max(1, self.number_of_sentences)
+    @staticmethod
+    def create_empty(tgt_language: str) -> TagMetric:
+        return TagMetric(
+            num_ref_tags=0,
+            num_correct_tags=0,
+            character_difference=0,
+            num_inconsistent_hyp=0,
+            num_inconsistent_tag_count=0,
+            num_inconsistent_text=0,
+            num_tags_inconsistent_sentences=0,
+            num_sentences=0,
+            tgt_language=tgt_language,
+        )
+
+    @staticmethod
+    def create_inconsistent(number_of_tags_in_sentence: int, tgt_language: str, inconsistency_type: InconsistencyType) -> TagMetric:
+        metric = TagMetric.create_empty(tgt_language=tgt_language)
+        metric.num_sentences = 1
+        metric.num_tags_inconsistent_sentences = number_of_tags_in_sentence
+        if inconsistency_type is InconsistencyType.HYPOTHESIS:
+            metric.num_inconsistent_hyp = 1
+        elif inconsistency_type is InconsistencyType.TAG_COUNT:
+            metric.num_inconsistent_tag_count = 1
+        elif inconsistency_type is InconsistencyType.TEXT:
+            metric.num_inconsistent_text = 1
+        else:
+            raise ValueError(f"Unsupported {inconsistency_type=}")
+
+        return metric
 
     def __add__(self, other: TagMetric) -> TagMetric:
         return TagMetric(
-            self.number_of_tags + other.number_of_tags,
-            self.number_of_correct_tags + other.number_of_correct_tags,
-            self.character_difference + other.character_difference,
-            self.number_of_inconsistent_sentences + other.number_of_inconsistent_sentences,
-            self.number_of_tags_in_inconsistent_sentences +
-            other.number_of_tags_in_inconsistent_sentences,
-            self.number_of_sentences + other.number_of_sentences,
+            num_ref_tags=self.num_ref_tags + other.num_ref_tags,
+            num_correct_tags=self.num_correct_tags + other.num_correct_tags,
+            character_difference=self.character_difference + other.character_difference,
+            num_inconsistent_hyp=self.num_inconsistent_hyp + other.num_inconsistent_hyp,
+            num_inconsistent_tag_count=self.num_inconsistent_tag_count + other.num_inconsistent_tag_count,
+            num_inconsistent_text=self.num_inconsistent_text + other.num_inconsistent_text,
+            num_tags_inconsistent_sentences=self.num_tags_inconsistent_sentences + other.num_tags_inconsistent_sentences,
+            num_sentences=self.num_sentences + other.num_sentences,
+            tgt_language=self.tgt_language,
         )
```

Comment:
In case we extend this further in the future, we could try to make use of a Counter as this code tends to get verbose. Counters already implement __add__, e.g.:
```
>>> from collections import Counter
>>> Counter("aabbbccc") + Counter("bbbbbccccccdddddd")
Counter({'c': 9, 'b': 8, 'd': 6, 'a': 2})
```